cupyx/scipy/interpolate/_rgi.py

```diff
@@ -0,0 +1,347 @@
+__all__ = ['RegularGridInterpolator', 'interpn']
+
+import itertools
+import cupy as cp
+
+
+def _ndim_coords_from_arrays(points, ndim=None):
+    """
+    Convert a tuple of coordinate arrays to a (..., ndim)-shaped array.
+    """
+    if isinstance(points, tuple) and len(points) == 1:
+        # handle argument tuple
+        points = points[0]
+    if isinstance(points, tuple):
+        p = cp.broadcast_arrays(*points)
+        n = len(p)
+        for j in range(1, n):
+            if p[j].shape != p[0].shape:
+                raise ValueError(
+                    "coordinate arrays do not have the same shape")
+        points = cp.empty(p[0].shape + (len(points),), dtype=float)
+        for j, item in enumerate(p):
+            points[..., j] = item
+    else:
+        points = cp.asanyarray(points)
+        if points.ndim == 1:
+            if ndim is None:
+                points = points.reshape(-1, 1)
+            else:
+                points = points.reshape(-1, ndim)
+    return points
+
+
+class RegularGridInterpolator:
+    _ALL_METHODS = ["linear", "nearest"]
+
+    def __init__(self, points, values, method="linear", bounds_error=True,
+                 fill_value=cp.nan):
+        if method not in self._ALL_METHODS:
+            raise ValueError("Method '%s' is not defined" % method)
+
+        self.method = method
+        self.bounds_error = bounds_error
+
+        if not hasattr(values, 'ndim'):
+            # allow reasonable duck-typed values
+            values = cp.asarray(values)
```

Comment:
I think this can be omiitted: the main reason for this in scipy was to be able to deal with out-of-memory arrays (NedCDF files, hdf5 etc). Not sure if it's valuable for CuPy.

Reply:
Removing it would make the interpolation fail if the inputs are lists instead of cupy arrays.  This might actually be desired behavior because the other cupy functions seem to strictly accept only cupy arrays, not array-like objects or even scalars.  Can a cupy dev let us know what the desired behavior is?

Reply:
Basically, CuPy takes `cupy.ndarray`s where NumPy/SciPy take general array-like. We'd like to avoid implicit host-to-device sync and to make it responsibility of library uses to transfer data. (However, I know, the policy is not always alive across CuPy's APIs...)